include/dali/core/device_guard.h

```diff
@@ -35,6 +35,9 @@ class DLL_PUBLIC DeviceGuard {
   //         for device id < 0 it is no-op
   explicit DeviceGuard(int new_device);
   ~DeviceGuard();
+
+  bool has_old_context();
```

Comment:
This is a very generic utility. Don't add application specific features to it - if you add new features, try to make them as useful as they can be - e.g.:
```suggestion
  CUContext old_context() const noexcept { return old_context_; }
```
Also, since this function just gets one field, it's much better to keep it inline instead of making it a DLL-exported function, which is totally isolated and makes optimization across the call impossible, even with LTO.

Reply:
done